src/ebay-fake-menu-button/menu-button.tsx

```diff
@@ -1,35 +1,54 @@
-import React, { cloneElement, FC, useEffect, useRef, useState } from 'react'
+import React, { cloneElement, ComponentProps, FC, useEffect, useRef, useState } from 'react'
 import classnames from 'classnames'
 
 import { filterByType, findComponent } from '../common/component-utils'
-import { randomId } from '../common/random-id'
-import { EbayButton, EbayButtonProps, EbayFakeMenu, EbayFakeMenuItemProps } from '..'
-import { EbayFakeMenuButtonItem, EbayFakeMenuButtonLabel, EbayFakeMenuButtonSeparator } from '.'
 import { handleEscapeKeydown } from '../common/event-utils'
+import { randomId } from '../common/random-id'
+import { EbayButton, EbayButtonProps, EbayFakeMenu, EbayFakeMenuItemProps, EbayIconButton } from '..'
+import { EbayFakeMenuButtonIcon, EbayFakeMenuButtonItem, EbayFakeMenuButtonLabel, EbayFakeMenuButtonSeparator } from '.'
+
+export type EbayFakeMenuButtonVariant = 'overflow' | 'form' | 'button'
 
-export type MenuButtonProps = {
+export type EbayFakeMenuButtonProps = {
     a11yText?: string;
+    noToggleIcon?: boolean;
+    expanded?: boolean;
+    fixWidth?: boolean;
+    reverse?: boolean;
+    variant?: EbayFakeMenuButtonVariant;
     className?: string;
     onCollapse?: () => void;
     onExpand?: () => void;
     text?: string;
+    // todo: implement the following props
+    type?: 'radio' | 'checkbox';
 }
 
-const EbayMenuButton: FC<EbayButtonProps & MenuButtonProps> = ({
+type ButtonProps = Omit<EbayButtonProps, 'variant'> &
+    Omit<ComponentProps<'button'>, 'type'> &
+    ComponentProps<'a'>
+type Props = ButtonProps & EbayFakeMenuButtonProps
+
+const EbayMenuButton: FC<Props> = ({
     a11yText,
-    borderless,
-    children,
+    noToggleIcon,
+    fixWidth,
+    reverse,
+    variant,
+    expanded: defaultExpanded = false,
     className,
     onCollapse = () => {},
     onExpand = () => {},
     text = '',
+    children,
     ...rest
 }) => {
-    const [expanded, setExpanded] = useState<boolean|undefined>()
+    const [expanded, setExpanded] = useState(defaultExpanded)
     const [menuId, setMenuId] = useState<string|undefined>()
     const ref = useRef<HTMLButtonElement>()
 
-    const label = findComponent(children, EbayFakeMenuButtonLabel) || text
+    const icon = findComponent(children, EbayFakeMenuButtonIcon)
```

Comment:
See my previous comment about EbayMenuButtonIcon. I think we can remove EbayMenuButtonIcon and just use EbayIcon